windows/hcs/hcs.go

```diff
@@ -427,55 +481,42 @@ func (s *HCS) newHCSConfiguration() (*hcsshim.ContainerConfig, error) {
 	}
 
 	if len(configuration.Layers) > 0 {
-		di.HomeDir = filepath.Dir(s.conf.Layers[0])
+		di.HomeDir = filepath.Dir(conf.Layers[0])
 	}
 
 	// Windows doesn't support creating a container with a readonly
 	// filesystem, so always create a RW one
-	if err := hcsshim.CreateSandboxLayer(di, s.id, s.conf.Layers[0], s.conf.Layers); err != nil {
+	if err := hcsshim.CreateSandboxLayer(di, id, conf.Layers[0], conf.Layers); err != nil {
 		return nil, errors.Wrapf(err, "failed to create sandbox layer for %s: layers: %#v, driverInfo: %#v",
-			s.id, configuration.Layers, di)
-	}
-
-	configuration.LayerFolderPath = filepath.Join(di.HomeDir, s.id)
-	if err := ioutil.WriteFile(filepath.Join(s.stateDir, layerFile), []byte(configuration.LayerFolderPath), 0644); err != nil {
-		log.L.WithError(err).Warnf("failed to save active layer %s", configuration.LayerFolderPath)
+			id, configuration.Layers, di)
 	}
+	configuration.LayerFolderPath = filepath.Join(di.HomeDir, id)
 
-	err := hcsshim.ActivateLayer(di, s.id)
+	err := hcsshim.ActivateLayer(di, id)
 	if err != nil {
 		removeLayer(context.TODO(), configuration.LayerFolderPath)
 		return nil, errors.Wrapf(err, "failed to active layer %s", configuration.LayerFolderPath)
 	}
 
-	err = hcsshim.PrepareLayer(di, s.id, s.conf.Layers)
+	err = hcsshim.PrepareLayer(di, id, conf.Layers)
 	if err != nil {
 		removeLayer(context.TODO(), configuration.LayerFolderPath)
 		return nil, errors.Wrapf(err, "failed to prepare layer %s", configuration.LayerFolderPath)
 	}
 
-	volumePath, err := hcsshim.GetLayerMountPath(di, s.id)
+	volumePath, err := hcsshim.GetLayerMountPath(di, id)
 	if err != nil {
-		if err := hcsshim.DestroyLayer(di, s.id); err != nil {
-			log.L.Warnf("failed to DestroyLayer %s: %s", s.id, err)
+		if err := hcsshim.DestroyLayer(di, id); err != nil {
+			log.L.Warnf("failed to DestroyLayer %s: %s", id, err)
 		}
-		return nil, errors.Wrapf(err, "failed to getmount path for layer %s: driverInfo: %#v", s.id, di)
+		return nil, errors.Wrapf(err, "failed to getmount path for layer %s: driverInfo: %#v", id, di)
 	}
 	configuration.VolumePath = volumePath
 
-	f, err := os.OpenFile(fmt.Sprintf("%s-hcs.json", s.id), os.O_CREATE|os.O_WRONLY|os.O_TRUNC, 066)
```

Comment:
Forgive my limited understanding of this code. Was this only used by `hcsshim` prior? I see it used to be written and never read under these packages. Is the replaced by the OCI runtime spec?